src/Functions/FunctionsConversion.h

```diff
@@ -1772,6 +1772,9 @@ class FunctionConvert : public IFunction
             }
         }
 
+        if (std::is_same_v<ToDataType, DataTypeString> && from_type->getCustomSerialization())
```

Comment:
Consider to add `if constexpr (std::is_same_v<ToDataType, DataTypeString>)` before this check